File: Programing/CollectionControl/Exercise1/AnswerCSharp/Exercise1.cs
using System.Collections.Generic;
using System.Linq;

namespace Exercise1
{
    class Exercise1
    {
        public int SumAgeOfOldestUser3(List<User> users)
	{
	    return users
	        .Select(user => user.Age)
		.OrderByDescending(age => age)
		.Take(3)
		.Sum();
	}
    }
}

File: Programing/CollectionControl/Exercise1/AnswerJava/Exercise1.java
import java.util.*;
import java.util.stream.*;

class Exercise1 {
    public int sumAgeOfOldestUser3(List<User> users) {
	return users.stream()
	    .map(user -> user.getAge())
	    .sorted(Comparator.reverseOrder())
	    .limit(3)
	    .reduce(0, (sum, age) -> sum + age);
    }
}
